roles/openshift_health_checker/openshift_checks/efk_stack.py

```diff
@@ -0,0 +1,604 @@
+# pylint: disable=missing-docstring
+'''
+  Module for performing checks on an elasticseatch, fluentd, and kibana stack
+'''
+
+import json
+import os
+import re
+import ssl
+import urllib2
+
+from openshift_checks import OpenShiftCheck, OpenShiftCheckException, get_var
+
+
+class EFKStack(OpenShiftCheck):
+    '''Module that checks an EFK stack shipped with OCP'''
+
+    name = "efk_stack"
+    tags = ["health"]
+
+    logging_namespace = None
+
+    @classmethod
+    def is_active(cls, task_vars):
+        '''Run only on first master and only when logging is configured. Returns: bool'''
+        logging_deployed = get_var(task_vars, "openshift_hosted_logging_deploy", default=False)
+        # Note: It would be nice to use membership in oo_first_master group, however for now it
+        # seems best to avoid requiring that setup and just check this is the first master.
+        hostname = get_var(task_vars, "ansible_ssh_host") or [None]
+        masters = get_var(task_vars, "groups", "masters", default=None) or [None]
+        is_active = logging_deployed and masters[0] == hostname
+        return super(EFKStack, cls).is_active(task_vars) and is_active
+
+    def run(self, tmp, task_vars):
+        '''Check various things and gather errors. Returns: result as hash'''
+        self.logging_namespace = get_var(task_vars, "openshift_logging_namespace", default="logging")
+        pods_for, error = self._get_pods(task_vars)
+        if error:
+            return {"failed": True, "changed": False, "msg": error}
+        check_error = dict(
+            elasticsearch=self.check_elasticsearch(pods_for['es'], task_vars),
+            fluentd=self.check_fluentd(pods_for['fluentd'], task_vars),
+            curator=self.check_curator(pods_for['curator']),
+            kibana=self.check_kibana(pods_for['kibana']),
```

Comment:
There aren't any dependencies between the methods for the different components at this time, so yeah, I suppose they could be split into separate checks for isolation/concurrency. There are a lot of module executions in this check and making those parallel could speed this up noticeably.